frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -191,39 +191,49 @@ local footerTextGeneratorMap = {
                 return ("%s / %s"):format(footer.ui.pagemap:getCurrentPageLabel(true),
                                           footer.ui.pagemap:getLastPageLabel(true))
             end
-            return ("%d / %d"):format(footer.pageno, footer.pages)
+            if footer.ui.document:hasFlows() then
+                -- i.e., if we are hiding non-linear fragments and there's anything to hide,
+                if footer.flow == 0 then
+                    return ("%d // %d"):format(footer.pageno, footer.pages)
+                else
+                    return ("[%d / %d]%d"):format(footer.pageno, footer.pages, footer.flow)
+                end
+            else
+                return ("%d / %d"):format(footer.pageno, footer.pages)
+            end
         elseif footer.position then
             return ("%d / %d"):format(footer.position, footer.doc_height)
         end
     end,
     pages_left = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].pages_left
-        local left = footer.ui.toc:getChapterPagesLeft(footer.pageno)
+        local left = footer.ui.toc:getChapterPagesLeft(footer.ui:getCurrentPage())
```

Comment:
Why that change for this one - and not for the others (like the one above you modified)?
Can footer.pageno be different from footer.ui:getCurrentPage() ?

Reply:
> Can footer.pageno be different from footer.ui:getCurrentPage() ?

Yes, `footer.pageno` is the page number shown in the footer, the "local" page in the current flow, while `footer.ui:getCurrentPage()` is the global page number, counting all flows.